### DISCUSSION TITLE
GodMode9 v2.1.1 Sixth Anniversary Release

### DISCUSSION THREAD

**Discussion by d0k3** [d0k3/GodMode9#767] [General]
Is it really that time of the year again? Six years ago, on March 22nd of the faraway year of 2016, GodMode9 was first released to the public. It has come a long way, reaching high stability and amassing features. Right now, there's not much left that GodMode9 can't do, and there's only the odd bug coming in from time to time. As such, this release is a rather small one, with only bugfixes and maintenance stuff.

I still wanted to celebrate the day with a release, so here is GodMode9 v2.1.1 with these changes:
* [improved] Improved installation of DSiWare games (thanks @Epicpkmn11)
* [scripting] Faster script loading as well as other improvements and fixes (thanks @aspargas2)
* [fixed] Fixed a nasty bug when trimming certain DS cart dumps
* [fixed] Some smaller bugfixes and general code maintenance

**How do I update GodMode9?**
(This may be the last time I tell you how to in the release notes, so please remember) You wouldn't believe how often we get that question when we do a new release. It's actually very simple: Just replace `GodMode9.firm` on your SD card with the file found in the release ZIP. You may also want to update scripts, which are found in the `./gm9` folder inside the archive and go to the same folder on your SD card.

**Special thanks**
I'll be honest, I've long given up mentioning everyone here. The sheer number of contributors (developers, people who report bugs or suggest features, people who help with advice or web hosting...) after 6 years of development is just too big. Just know, each and every contribution, no matter how big or small is highly appreciated. For this release, special thanks go to @Wolfvak, who's been a main dev with me for almost all of the lifetime of GodMode9, to @aspargas2 who did maintenance and fixed bugs, to @Epicpkmn11, who improved the save game generation and thus the installation for DSiWare CIAs and dumps and to @giiutfff who catched a really nasty bug happening when trimming certain DS cart dumps in GodMode9 (great catch!).

<hr /><em>This discussion was created from the release <a href='https://github.com/d0k3/GodMode9/releases/tag/v2.1.1'>GodMode9 v2.1.1 Sixth Anniversary Release</a>.</em>

**Comment by strobbery707** [comment]
howdy y'all!! i'm here from 3ds hacks guide, tryna mod a 2ds. whenever i download ur zip my free antivirus stops me saying there's a trojan! she says it's name is Trojan.GenericKD.66205510 .  just thought u should know! have a good one :3

**Reply by bompf** [reply]
same here, can't open the file

**Reply by Epicpkmn11** [reply]
There's not even any Windows executables, so it's clearly a false positive. Just look for the ignore button or temporarily turn off your antivirus and you should be good.

**Reply by strobbery707** [reply]
It’s not in the actual zip itself, the antivirus blocks it before I even chose where to save it in explorer. I did pause my av and download it, and then did a full scan of my system and network and nothing popped up. So hopefully nothing to worry about. :-)

**Comment by tr3bleclef** [comment]
hiya, came here from the 3ds hacks guide too, and i may sound way too dumb but its my first time doing this and i can't find the godmode9.firm file anywhere :( please help, is it in a folder?

**Reply by Epicpkmn11** [reply]
Did you perhaps download the source code by mistake? You want 
[GodMode9-v2.1.1-20220322194259.zip](https://github.com/d0k3/GodMode9/releases/download/v2.1.1/GodMode9-v2.1.1-20220322194259.zip)

With the right zip it'll just be right inside the zip

**Reply by tr3bleclef** [reply]
thats so weird, i downloaded the zip you just replied with and it opened a different zip than the one i have, but found it! thank you so much <3

**Reply by Cyben2004** [reply]
I need help with downloading to sd card it says download failed or not complete so i just need some simple help(Im not angry Im just kinda disappointed that it does not work for me) Id like some help thank you in advance :)

**Reply by Cyben2004** [reply]
nvermind i dont need help any more but i will contact you guys if there is any problems

**Comment by trulyfloss** [comment]
hi im trying to do the 3ds guide its my first time doing this and every time i download the zip it and i open it it says file cannot be found what do i do?

**Reply by Epicpkmn11** [reply]
It downloads fine for me... Make sure you're downloading [GodMode9-v2.1.1-20220322194259.zip](https://github.com/d0k3/GodMode9/releases/download/v2.1.1/GodMode9-v2.1.1-20220322194259.zip)

**Comment by lilpenut** [comment]
hi there im trying to download the zip file but I'm getting messages like "failed - insufficient permissions" or "failed - download error" I would really appreciate some guidance on how to successfully download the file

**Reply by Mikuelol** [reply]
same for me :c

**Comment by BabaYaga3001** [comment]
Its full of virus. Both my antivirus is blocking the file. Its Trojan.GenericKD.662055. Stop screwing everyone around with this crap

**Reply by Epicpkmn11** [reply]
Your antivirus is throwing a false positive, there's not even any Windows excecutables inside

**Comment by Danterfdfdfd** [comment]
I'm having trouble downloading the zip file, it jus keeps saying "Failed - Network Error" i shut down my antivirus but that didn't work either, any ideas on what's happening?

**Comment by sandcr** [comment]
Hey there, I am trying to mod my 3ds and whenever I go to download GM9, a "Failed - Network error" or "Failed - Needs authorization" message appears, any ideas on how to fix this

**Reply by Glade061** [reply]
i am having the exact same issue
maybe it has something to do with windows 11 is what im thinking tho

**Reply by urherenow** [reply]
Nope. Just downloaded it on 2 windows 11 machines, one with Chrome, and the other through IDM (via chrome). It's either an Antivirus/antimalware issue, an ISP issue, or a VPN issue (blocks of VPN IP addresses may get blocked by sites like github because of absusers, etc...)

**Comment by Glade061** [comment]
no matter what i do it doesnt let me download this andit keeps saying im having network issues even though i have full bars of wi-fi

**Reply by urherenow** [reply]
See my reply above. It is indeed an issue on your end.

**Reply by Glade061** [reply]
i dont even have a vpn and its saying its a network error

**Reply by urherenow** [reply]
Then try using one. Or use a different device, like your phone. I'm telling you the problem isn't with GitHub.

**Comment by Cyben2004** [comment]
Does this work for the original 3ds?

**Reply by Cyben2004** [reply]
Im Kinda new to modding

**Comment by Cyben2004** [comment]
I can only contact you while im at summer school or off my moms computer I know the password but i can only do it while my mom is at work i will be able to be contacted all this week until saturday im using my school account just to let you know and on wednesdays my mom is off work so this is the only time im able to talk only when in summer school on wednesday and my summer school ends at the end of this month.

**Comment by ssstrab3wwy** [comment]
Hi! so i've recently decided to mod my old New 3ds XL, and everything has gone fine, until I have to boot GodMode9. For some reason, my d pad gets stuck going up (I have checked, and it only happens in gm9) and I can't do the verification code :(

**Comment by keybladekallen** [comment]
Whenever i try to load GodMode9 it always gives me the "payloads are corrupted" error and when i tried to get a newer version of the .firm file it would only copy over an empty file im not sure what i did wrong can anyone help?
